filterapi/filterconfig.go

```diff
@@ -156,15 +156,55 @@ type RouteRule struct {
 // besides that this abstracts the concept of a backend at Envoy Gateway level to a simple name.
 type Backend struct {
 	// Name of the backend, which is the value in the final routing decision
-	// matching the header key specified in the [Config.BackendRoutingHeaderKey].
+	// matching the header key specified in the [filterapi.Config.BackendRoutingHeaderKey].
 	Name string `json:"name"`
 	// Schema specifies the API schema of the output format of requests from.
 	Schema VersionedAPISchema `json:"schema"`
 	// Weight is the weight of the backend in the routing decision.
+	//
+	// When DynamicLoadBalancing is specified, the weight is ignored.
 	Weight int `json:"weight"`
 	// Auth is the authn/z configuration for the backend. Optional.
-	// TODO: refactor after https://github.com/envoyproxy/ai-gateway/pull/43.
 	Auth *BackendAuth `json:"auth,omitempty"`
+	// DynamicLoadBalancing is the dynamic backend configuration which forces the AI filter to
+	// route the request to a specific endpoint (ip:port) instead of the normal cluster routing.
+	//
+	// When this is specified, the AI Gateway filter assume that the ORIGINAL_DST cluster is configured
+	// to route the request to an endpoint from `x-ai-eg-original-dst` header.
+	DynamicLoadBalancing *DynamicLoadBalancing `json:"dynamicLoadBalancing,omitempty"`
+}
+
+// DynamicLoadBalancing corresponds to InferencePool and InferenceModels belonging to the same pool.
+type DynamicLoadBalancing struct {
+	// Models that can be served by this backend. If not matched, the 404 is returned to the client.
+	//
+	// If multiple models are provided, the request is routed to the backend based on the weights, criticality, etc.
```

Comment:
We also want fallback between multiple models, for example if `provisioned throughout` model does not have extra capacity we fallback to `on-demand` model. In this case routing is based on model name. A concrete example is to route to `bedrock-runtime.us-east-1.aws.com` and fallback using two different model names `model1-pt` and `model1-od`.

Reply:
We may need to extend InferenceModel API to add a field `fallbackModels`

Reply:
Is Models and Backends are 1-1 maping, like
model[i] -> backend[i] ?

or 
[mode0, model1, ... model_N-1] -> backend [0]
...
[mode0, model1, ... model_N-1] -> backend [N-1]

Reply:
this case N to M is the answer. InferencePool can reference multiple AIServiceBackends via Selector and one pool can have multiple models

Reply:
so the model name fallback is interesting ... I think it's not within the realm of InferencePool at the moment vs currently it only has weight based random picking (optional) cc @kfswain https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/a13a1239330ffaaafa4d0f948c00cafd106086aa/pkg/epp/handlers/request.go#L71-L72